### PR TITLE
Add KPrepublic BM60hsrgb_iso/rev2, acting in lieu of @bdtc123

### PR DESCRIPTION
**Pull request by peepeetee**
## Description
I've been asked by @bdtc123 to upload their code to qmk, and this is one of the keyboards.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #14964

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kprepublic/bm60v2_iso/bm60v2_iso.c [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "bm60v2_iso.h"
+
+#ifdef RGB_MATRIX_ENABLE
+const is31_led g_is31_leds[DRIVER_LED_TOTAL] = {
```
Comment (drashna) [inline keyboards/kprepublic/bm60v2_iso/bm60v2_iso.c:19]:
```suggestion
const __flash is31_led g_is31_leds[DRIVER_LED_TOTAL] = {
```

File: keyboards/kprepublic/bm60v2_iso/config.h [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4B50
+#define PRODUCT_ID      0x1123
+#define DEVICE_VER      0x0002
+#define MANUFACTURER    KP REPUBLIC
+#define PRODUCT         BM60V2_ISO
+#define DESCRIPTION     BM60V2_ISO RGB keyboard
```
Comment (drashna) [inline keyboards/kprepublic/bm60v2_iso/config.h:26]:
```suggestion
```

File: keyboards/kprepublic/bm60v2_iso/config.h [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+#define DEBOUNCE 3
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+//rgb light setting
+#define RGBLIGHT_LIMIT_VAL 150
+#define RGBLED_NUM          6
+#define RGB_DI_PIN          F0
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/kprepublic/bm60v2_iso/config.h:60]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/kprepublic/bm60v2_iso/info.json [hunk 4]
```diff
@@ -0,0 +1,81 @@
+{
+    "keyboard_name": "bm60v2_ISO",
+    "url": "",
+    "maintainer": "kp republic",
+    "width": 15,
+    "height": 5,
```
Comment (drashna) [inline keyboards/kprepublic/bm60v2_iso/info.json:6]:
```suggestion
```

File: keyboards/kprepublic/bm60v2_iso/rules.mk [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
```
Comment (drashna) [inline keyboards/kprepublic/bm60v2_iso/rules.mk:12]:
```suggestion
```

File: keyboards/kprepublic/bm60v2_iso/rules.mk [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite        # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/kprepublic/bm60v2_iso/rules.mk:17]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/kprepublic/bm60v2_iso/rules.mk [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes           # Enable keyboard RGB underglow
+MIDI_ENABLE = no               # MIDI support
+BLUETOOTH_ENABLE = no          # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no              # Audio output on port C6
+RGB_MATRIX_ENABLE = yes        # Use RGB matrix
+RGB_MATRIX_DRIVER = IS31FL3733
+NO_USB_STARTUP_CHECK = no      # Disable initialization only when usb is plugged in
```
Comment (drashna) [inline keyboards/kprepublic/bm60v2_iso/rules.mk:33]:
```suggestion
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes              # USB Nkey Rollover
BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes          # Enable keyboard RGB underglow
AUDIO_ENABLE = no              # Audio output
RGB_MATRIX_ENABLE = yes        # Use RGB matrix
RGB_MATRIX_DRIVER = IS31FL3733
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kprepublic/bm60v2_iso/bm60v2_iso.c [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+__attribute__ ((weak))
+void rgb_matrix_indicators_user(void) {
+    if (host_keyboard_led_state().caps_lock) {
+        rgb_matrix_set_color(28, 0xFF, 0xFF, 0xFF);
+    }
+}
+#endif
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
```
Comment (drashna) [inline keyboards/kprepublic/bm60v2_iso/bm60v2_iso.c:129]:
```suggestion
    if (!process_record_user(keycode, record)) { return false; }
    
    if (record->event.pressed) {
```

File: keyboards/kprepublic/bm60v2_iso/config.h [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+#    define DISABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
+#    define DISABLE_RGB_MATRIX_BAND_SPIRAL_SAT
+#    define DISABLE_RGB_MATRIX_RAINDROPS
+#    define DISABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
+#    define DRIVER_ADDR_1 0b1010000
+#    define DRIVER_ADDR_2 0b1010000 // this is here for compliancy reasons.
+#    define DRIVER_COUNT 2
+#    define DRIVER_1_LED_TOTAL 64
+#    define DRIVER_LED_TOTAL DRIVER_1_LED_TOTAL
+
```
Comment (drashna) [inline keyboards/kprepublic/bm60v2_iso/config.h:87]:
```suggestion

#define ENABLE_RGB_MATRIX_ALPHAS_MODS
#define ENABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#define ENABLE_RGB_MATRIX_GRADIENT_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_BREATHING
#define ENABLE_RGB_MATRIX_BAND_SAT
#define ENABLE_RGB_MATRIX_BAND_VAL
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
#define ENABLE_RGB_MATRIX_CYCLE_ALL
#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
#define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
#define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
#define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
#define ENABLE_RGB_MATRIX_DUAL_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
#define ENABLE_RGB_MATRIX_RAINDROPS
#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
#define ENABLE_RGB_MATRIX_HUE_BREATHING
#define ENABLE_RGB_MATRIX_HUE_PENDULUM
#define ENABLE_RGB_MATRIX_HUE_WAVE
#define ENABLE_RGB_MATRIX_FRACTAL
#define ENABLE_RGB_MATRIX_PIXEL_RAIN

#define ENABLE_RGB_MATRIX_TYPING_HEATMAP
#define ENABLE_RGB_MATRIX_DIGITAL_RAIN

#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#define ENABLE_RGB_MATRIX_SPLASH
#define ENABLE_RGB_MATRIX_MULTISPLASH
#define ENABLE_RGB_MATRIX_SOLID_SPLASH
#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH

```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/kprepublic/bm60hsrgb_iso/rev2/rules.mk [hunk 10]
```diff
@@ -0,0 +1,38 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_iso/rev2/rules.mk:4]: ?

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kprepublic/bm60hsrgb_iso/rev2/rules.mk (hunk 10 above)
Comment (peepeetee) [reply keyboards/kprepublic/bm60hsrgb_iso/rev2/rules.mk:4]: ?

Comment (fauxpark) [reply keyboards/kprepublic/bm60hsrgb_iso/rev2/rules.mk:4]: There's no bootloader specified.

Comment (peepeetee) [reply keyboards/kprepublic/bm60hsrgb_iso/rev2/rules.mk:4]: What should I do?

Comment (fauxpark) [reply keyboards/kprepublic/bm60hsrgb_iso/rev2/rules.mk:4]: Specify one? I'm guessing it's probably `atmel-dfu`, but you should double check.

Comment (peepeetee) [reply keyboards/kprepublic/bm60hsrgb_iso/rev2/rules.mk:4]: image
facepalm.jpg

Comment (peepeetee) [reply keyboards/kprepublic/bm60hsrgb_iso/rev2/rules.mk:4]: fixed

File: keyboards/kprepublic/bm60hsrgb_iso/rev2/config.h [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+#    define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#endif
+
+
+
+
+
+
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_iso/rev2/config.h:170]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_iso/rev2/rev2.h [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+) { \
+    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D   }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C, KC_NO   }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39, KC_NO,   K3B,   K3C,   K3D   }, \
+    { K40,   K41,   K42,   K43,   KC_NO, K45, KC_NO,   KC_NO, KC_NO, K49,   K4A,   K4B,   K4C,   K4D   }, \
+}
+
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_iso/rev2/rev2.h:35]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_iso/rev2/rules.mk [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+
+
+# RGB Matrix is required to support per-key LEDs connected to IS31FL3733.
+RGB_MATRIX_ENABLE = yes
+
+# The simpler RGB Lighting code can be enabled to control just the underglow
+# LEDs from it; leave it disabled to make the RGB Matrix code control all LEDs,
+# including underglow.
+RGBLIGHT_ENABLE = no
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_iso/rev2/rules.mk:28]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kprepublic/bm60hsrgb_iso/readme.md [hunk 14]
```diff
@@ -4,16 +4,20 @@
...
-    make kprepublic/bm60hsrgb_iso/rev1:default
+    for rev1: make kprepublic/bm60hsrgb_iso/rev1:default
+    for rev2: make kprepublic/bm60hsrgb_iso/rev2:default
 
 Flashing example for this keyboard:
 
-    make kprepublic/bm60hsrgb_iso:default:flash
+    for rev1: make kprepublic/bm60hsrgb_iso/rev1:default:flash
+    for rev2: make kprepublic/bm60hsrgb_iso/rev2:default:flash
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_iso/readme.md:19]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_iso/rev2/config.h [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+// RGB_MATRIX_ENABLE (usually this is not required, but with the custom
+// IS31FL3733+WS2812 driver setup used by this board the LED controller drivers
+// are compiled unconditionally).
+
+#define ISSI_LED_TOTAL DRIVER_1_LED_TOTAL
+
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 180
+
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_iso/rev2/config.h:147]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_iso/rev2/config.h [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#define ENABLE_RGB_MATRIX_SPLASH
+#define ENABLE_RGB_MATRIX_MULTISPLASH
+#define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_iso/rev2/config.h:135]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_iso/rev2/rev2.c [hunk 17]
```diff
@@ -0,0 +1,207 @@
...
+
+#ifdef RGB_MATRIX_ENABLE
+
+
+
+#    include "i2c_master.h"
+#    include "is31fl3733.h"
+#    include "ws2812.h"
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_iso/rev2/rev2.c:28]:
```suggestion
#    include "i2c_master.h"
#    include "is31fl3733.h"
#    include "ws2812.h"

```

File: keyboards/kprepublic/bm60hsrgb_iso/rev2/rev2.c [hunk 18]
```diff
@@ -0,0 +1,207 @@
...
+__attribute__ ((weak))
+void rgb_matrix_indicators_user(void) {
+    if (host_keyboard_led_state().caps_lock) {
+        rgb_matrix_set_color(28, 0xFF, 0xFF, 0xFF);
+    }
+}
+
+
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_iso/rev2/rev2.c:145]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_iso/rev2/rev2.c [hunk 19]
```diff
@@ -0,0 +1,207 @@
...
+    .init          = rgb_matrix_driver_init,
+    .flush         = rgb_matrix_driver_flush,
+    .set_color     = rgb_matrix_driver_set_color,
+    .set_color_all = rgb_matrix_driver_set_color_all,
+};
+// clang-format on
+
+#endif /* RGB_MATRIX_ENABLE */
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_iso/rev2/rev2.c:207]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**